Launcher runs fail with EdcInjectionException (IntelliJ version)

# Bug Report

## Describe the Bug

Trying to run any luncher but the identity-hubs in IntelliJ results in an `EdcInjectionException` with a lot of missing injected fields. Identity-hubs start without errors. **Important**: I'm very new to this and even to gradle so maybe I do something fundamentelly wrong.

_A clear and concise description of the bug._

Running `:launchers:catalog-server:run` results in:

```cmd
SEVERE 2024-10-24T13:38:13.5831497 Error booting runtime: The following injected fields were not provided: Field "jtiValidationStore" of type [interface org.eclipse.edc.jwt.validation.jti.JtiValidationStore] required by org.eclipse.edc.iam.identitytrust.core.DcpDefaultServicesExtension Field "externalSigner" of type [interface org.eclipse.edc.jwt.signer.spi.JwsSignerProvider] required by org.eclipse.edc.iam.identitytrust.core.DcpDefaultServicesExtension Field "rulesRegistry" of type [interface org.eclipse.edc.token.spi.TokenValidationRulesRegistry] required by org.eclipse.edc.iam.identitytrust.core.IdentityAndTrustExtension Field "tokenValidationService" of type [interface org.eclipse.edc.token.spi.TokenValidationService] required by org.eclipse.edc.iam.identitytrust.core.IdentityAndTrustExtension Field "jtiValidationStore" of type [interface org.eclipse.edc.jwt.validation.jti.JtiValidationStore] required by org.eclipse.edc.iam.identitytrust.core.IdentityAndTrustExtension
Exception in thread "main" org.eclipse.edc.spi.EdcException: org.eclipse.edc.boot.system.injection.EdcInjectionException: The following injected fields were not provided:
org.eclipse.edc.boot.system.injection.EdcInjectionException: The following injected fields were not provided:
Field "jtiValidationStore" of type [interface org.eclipse.edc.jwt.validation.jti.JtiValidationStore] required by org.eclipse.edc.iam.identitytrust.core.DcpDefaultServicesExtension
Field "jtiValidationStore" of type [interface org.eclipse.edc.jwt.validation.jti.JtiValidationStore] required by org.eclipse.edc.iam.identitytrust.core.DcpDefaultServicesExtension
Field "externalSigner" of type [interface org.eclipse.edc.jwt.signer.spi.JwsSignerProvider] required by org.eclipse.edc.iam.identitytrust.core.DcpDefaultServicesExtension
Field "externalSigner" of type [interface org.eclipse.edc.jwt.signer.spi.JwsSignerProvider] required by org.eclipse.edc.iam.identitytrust.core.DcpDefaultServicesExtension
Field "rulesRegistry" of type [interface org.eclipse.edc.token.spi.TokenValidationRulesRegistry] required by org.eclipse.edc.iam.identitytrust.core.IdentityAndTrustExtension
Field "tokenValidationService" of type [interface org.eclipse.edc.token.spi.TokenValidationService] required by org.eclipse.edc.iam.identitytrust.core.IdentityAndTrustExtension
Field "jtiValidationStore" of type [interface org.eclipse.edc.jwt.validation.jti.JtiValidationStore] required by org.eclipse.edc.iam.identitytrust.core.IdentityAndTrustExtension
	at org.eclipse.edc.boot.system.DependencyGraph.of(DependencyGraph.java:102)
	at org.eclipse.edc.boot.system.ExtensionLoader.loadServiceExtensions(ExtensionLoader.java:127)
	at org.eclipse.edc.boot.system.runtime.BaseRuntime.createExtensions(BaseRuntime.java:163)
	at org.eclipse.edc.boot.system.runtime.BaseRuntime.boot(BaseRuntime.java:91)
Field "rulesRegistry" of type [interface org.eclipse.edc.token.spi.TokenValidationRulesRegistry] required by org.eclipse.edc.iam.identitytrust.core.IdentityAndTrustExtension
Field "tokenValidationService" of type [interface org.eclipse.edc.token.spi.TokenValidationService] required by org.eclipse.edc.iam.identitytrust.core.IdentityAndTrustExtension
Field "jtiValidationStore" of type [interface org.eclipse.edc.jwt.validation.jti.JtiValidationStore] required by org.eclipse.edc.iam.identitytrust.core.IdentityAndTrustExtension
	at org.eclipse.edc.boot.system.runtime.BaseRuntime.main(BaseRuntime.java:77)
	at org.eclipse.edc.boot.system.runtime.BaseRuntime.onError(BaseRuntime.java:143)
	at org.eclipse.edc.boot.system.runtime.BaseRuntime.boot(BaseRuntime.java:106)
	at org.eclipse.edc.boot.system.runtime.BaseRuntime.main(BaseRuntime.java:77)
Caused by: org.eclipse.edc.boot.system.injection.EdcInjectionException: The following injected fields were not provided:
Field "jtiValidationStore" of type [interface org.eclipse.edc.jwt.validation.jti.JtiValidationStore] required by org.eclipse.edc.iam.identitytrust.core.DcpDefaultServicesExtension
Field "externalSigner" of type [interface org.eclipse.edc.jwt.signer.spi.JwsSignerProvider] required by org.eclipse.edc.iam.identitytrust.core.DcpDefaultServicesExtension
Field "rulesRegistry" of type [interface org.eclipse.edc.token.spi.TokenValidationRulesRegistry] required by org.eclipse.edc.iam.identitytrust.core.IdentityAndTrustExtension
Field "tokenValidationService" of type [interface org.eclipse.edc.token.spi.TokenValidationService] required by org.eclipse.edc.iam.identitytrust.core.IdentityAndTrustExtension
Field "jtiValidationStore" of type [interface org.eclipse.edc.jwt.validation.jti.JtiValidationStore] required by org.eclipse.edc.iam.identitytrust.core.IdentityAndTrustExtension
	at org.eclipse.edc.boot.system.DependencyGraph.of(DependencyGraph.java:102)
	at org.eclipse.edc.boot.system.ExtensionLoader.loadServiceExtensions(ExtensionLoader.java:127)
	at org.eclipse.edc.boot.system.runtime.BaseRuntime.createExtensions(BaseRuntime.java:163)
	at org.eclipse.edc.boot.system.runtime.BaseRuntime.boot(BaseRuntime.java:91)
	... 1 more

Caused by: org.eclipse.edc.boot.system.injection.EdcInjectionException: The following injected fields were not provided:
```

### Expected Behavior

Sub-modules start.

### Observed Behavior

Sub-modules don't start beacuse of unprovided injections. See above.

## Steps to Reproduce

1. Clone repo
2. `gradle build`
3. `gradle run --scan` or run a sub-module (launcher) other than identiy-hubs

## Context Information

- Used version: 0.10.0-SNAPSHOT
- OS: Windows 11 (sorry, can't get WSL to run atm)
- I'm very new to this and even to gradle so maybe I do something fundamentelly wrong.

--- a/deployment/modules/identity-hub/main.tf
+++ b/deployment/modules/identity-hub/main.tf
@@ -163,6 +163,7 @@ resource "kubernetes_config_map" "identityhub-config" {
     EDC_SQL_SCHEMA_AUTOCREATE              = true
     EDC_STS_ACCOUNT_API_URL                = var.sts-accounts-api-url
     EDC_STS_ACCOUNTS_API_AUTH_HEADER_VALUE = "password"
+    EDC_IAM_ACCESSTOKEN_JTI_VALIDATION     = true
   }
 }
 

--- a/deployment/provider.tf
+++ b/deployment/provider.tf
@@ -56,9 +56,9 @@ module "provider-identityhub" {
   namespace         = kubernetes_namespace.ns.metadata.0.name
 
   database = {
-    user     = "identityhub"
-    password = "identityhub"
-    url      = "jdbc:postgresql://${module.provider-postgres.database-url}/identityhub"
+    user     = "identity"
+    password = "identity"
+    url      = "jdbc:postgresql://${module.provider-postgres.database-url}/identity"
   }
   sts-accounts-api-url = module.provider-sts.sts-accounts-url
 }
@@ -70,9 +70,9 @@ module "provider-sts" {
   humanReadableName = "provider-sts"
   namespace         = kubernetes_namespace.ns.metadata.0.name
   database = {
-    user     = "sts"
-    password = "sts"
-    url      = "jdbc:postgresql://${module.provider-postgres.database-url}/sts"
+    user     = "identity"
+    password = "identity"
+    url      = "jdbc:postgresql://${module.provider-postgres.database-url}/identity"
   }
   vault-url = "http://provider-vault:8200"
 }
@@ -109,7 +109,6 @@ module "provider-postgres" {
     kubernetes_config_map.postgres-initdb-config-pqna.metadata[0].name,
     kubernetes_config_map.postgres-initdb-config-pm.metadata[0].name,
     kubernetes_config_map.postgres-initdb-config-ih.metadata[0].name,
-    kubernetes_config_map.postgres-initdb-config-sts.metadata[0].name
   ]
   namespace = kubernetes_namespace.ns.metadata.0.name
 }
@@ -166,23 +165,9 @@ resource "kubernetes_config_map" "postgres-initdb-config-ih" {
   }
   data = {
     "ih-initdb-config.sql" = <<-EOT
-        CREATE USER identityhub WITH ENCRYPTED PASSWORD 'identityhub' SUPERUSER;
-        CREATE DATABASE identityhub;
-        \c identityhub
-      EOT
-  }
-}
-
-resource "kubernetes_config_map" "postgres-initdb-config-sts" {
-  metadata {
-    name      = "sts-initdb-config"
-    namespace = kubernetes_namespace.ns.metadata.0.name
-  }
-  data = {
-    "sts-initdb-config.sql" = <<-EOT
-        CREATE USER sts WITH ENCRYPTED PASSWORD 'sts' SUPERUSER;
-        CREATE DATABASE sts;
-        \c sts
+        CREATE USER identity WITH ENCRYPTED PASSWORD 'identity' SUPERUSER;
+        CREATE DATABASE identity;
+        \c identity
       EOT
   }
 }

--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -26,6 +26,7 @@ edc-did-core = { module = "org.eclipse.edc:identity-did-core", version.ref = "ed
 edc-did-web = { module = "org.eclipse.edc:identity-did-web", version.ref = "edc" }
 edc-core-connector = { module = "org.eclipse.edc:connector-core", version.ref = "edc" }
 edc-core-crypto = { module = "org.eclipse.edc:crypto-core", version.ref = "edc" }
+edc-core-token = { module = "org.eclipse.edc:token-core", version.ref = "edc" }
 edc-core-edrstore = { module = "org.eclipse.edc:edr-store-core", version.ref = "edc" }
 edc-ext-http = { module = "org.eclipse.edc:http", version.ref = "edc" }
 edc-ext-jsonld = { module = "org.eclipse.edc:json-ld", version.ref = "edc" }
@@ -109,6 +110,7 @@ edc-sql-transactionlocal = { module = "org.eclipse.edc:transaction-local", versi
 edc-sql-accesstokendata = { module = "org.eclipse.edc:accesstokendata-store-sql", version.ref = "edc" }
 edc-sql-dataplane = { module = "org.eclipse.edc:data-plane-store-sql", version.ref = "edc" }
 edc-sql-dataplane-instancestore = { module = "org.eclipse.edc:data-plane-instance-store-sql", version.ref = "edc" }
+edc-sql-jtivdalidation = { module = "org.eclipse.edc:jti-validation-store-sql", version.ref = "edc" }
 
 
 # identity hub SQL implementations
@@ -176,7 +178,7 @@ parsson = { module = "org.eclipse.parsson:parsson", version.ref = "parsson" }
 [bundles]
 dpf = ["edc-dpf-selector-core", "edc-spi-dataplane-selector", "edc-dpf-selector-control-api", "edc-dpf-signaling-client", "edc-dpf-transfer-signaling"]
 
-connector = ["edc-boot", "edc-core-connector", "edc-ext-http", "edc-api-observability", "edc-ext-jsonld"]
+connector = ["edc-boot", "edc-core-connector", "edc-ext-http", "edc-api-observability", "edc-ext-jsonld", "edc-core-token"]
 
 controlplane = ["edc-controlplane-core", "edc-config-filesystem", "edc-auth-tokenbased", "edc-auth-configuration", "edc-api-management",
     "edc-api-management-config", "edc-api-management-edr", "edc-api-management-dataplaneselector",
@@ -196,9 +198,9 @@ dcp = ["edc-dcp", "edc-did-core", "edc-did-web", "edc-oauth2-client", "edc-dcp-c
 sql-edc = ["edc-sql-assetindex", "edc-sql-contractdef", "edc-sql-contractneg", "edc-sql-policydef", "edc-sql-edrcache", "edc-sql-transferprocess", "edc-sql-dataplane-instancestore", "edc-sql-core", "edc-sql-lease", "edc-sql-pool", "edc-sql-transactionlocal", "postgres"]
 sql-edc-dataplane = ["edc-sql-accesstokendata", "edc-sql-dataplane", "edc-sql-core", "edc-sql-lease", "edc-sql-pool", "edc-sql-transactionlocal", "edc-sql-dataplane-instancestore", "postgres"]
 
-sql-ih = ["edc-sql-ih-credstore-sql", "edc-sql-ih-didstore-sql", "edc-sql-ih-keypairstore-sql", "edc-sql-ih-pcstore-sql", "edc-sql-ih-stsstore-sql", "edc-sql-core", "edc-sql-pool", "edc-sql-transactionlocal", "postgres"]
+sql-ih = ["edc-sql-ih-credstore-sql","edc-sql-jtivdalidation", "edc-sql-ih-didstore-sql", "edc-sql-ih-keypairstore-sql", "edc-sql-ih-pcstore-sql", "edc-sql-ih-stsstore-sql", "edc-sql-core", "edc-sql-pool", "edc-sql-transactionlocal", "postgres"]
 
-sql-sts = [ "edc-sql-ih-stsstore-sql", "edc-sql-core", "edc-sql-pool", "edc-sql-transactionlocal", "postgres"]
+sql-sts = ["edc-sql-ih-stsstore-sql", "edc-sql-jtivdalidation", "edc-sql-core", "edc-sql-pool", "edc-sql-transactionlocal", "postgres"]
 
 sts = ["edc-sts-core", "edc-sts-api", "edc-sts-spi", "edc-sts"]
 

--- a/launchers/controlplane/build.gradle.kts
+++ b/launchers/controlplane/build.gradle.kts
@@ -27,6 +27,7 @@ dependencies {
     implementation(libs.bundles.controlplane)
     implementation(libs.bundles.dcp)
     implementation(libs.edc.core.connector)
+    implementation(libs.edc.core.token)
 
     if (project.properties.getOrDefault("persistence", "false") == "true") {
         runtimeOnly(libs.edc.vault.hashicorp)

